integration/simulation/transaction_injector.go

```diff
@@ -150,6 +155,55 @@ func (ti *TransactionInjector) Stop() {
 	}
 }
 
+// issueRandomValueTransfers creates and issues a number of L2 value transfer transactions proportional to the simulation time, such that they can be processed
+func (ti *TransactionInjector) issueRandomValueTransfers() {
+	for txCounter := 0; ti.shouldKeepIssuing(txCounter); txCounter++ {
+		fromWallet := ti.rndObsWallet()
+		toWallet := ti.rndObsWallet()
+		obscuroClient := ti.rpcHandles.ObscuroWalletRndClient(fromWallet)
+		// We avoid transfers to self, unless there is only a single L2 wallet.
+		for len(ti.wallets.SimObsWallets) > 1 && fromWallet.Address().Hex() == toWallet.Address().Hex() {
+			toWallet = ti.rndObsWallet()
+		}
+		toWalletAddr := toWallet.Address()
+		txData := &types.LegacyTx{
+			Nonce:    fromWallet.GetNonceAndIncrement(),
+			Value:    big.NewInt(int64(testcommon.RndBtw(1, 500))),
+			Gas:      uint64(1_000_000),
+			GasPrice: gethcommon.Big1,
+			To:       &toWalletAddr,
+		}
+		// tx := ti.newObscuroTransferTx(fromWallet, toWallet.Address(), testcommon.RndBtw(1, 500))
+		tx, err := obscuroClient.EstimateGasAndGasPrice(txData)
+		if err != nil {
+			panic(err)
```

Comment:
If the balance of the account is insufficient for the transfer, would the transaction get rejected and propagate an error? I imagine a wallet running out of eth is a possibility, so not sure if we want to panic.